presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -191,6 +188,194 @@ This table can then be queried in Presto::
+
+
+``hive.metastore.partition-batch-size.min``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``10``
+ * **Description:** See ``hive.metastore.partition-batch-size.max``.
+
+
+``hive.optimized-reader.enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** *Deprecated* Enables number of reader improvements introduced by alternative ORC implementation. The new reader supports vectorized reads, lazy loading, and predicate push down, all of which make the reader more efficient and typically reduces wall clock time for a query. However as the code has changed significantly it may or may not introduce some minor issues, so it can be disabled if some  problems with environment are noticed.
+
+
+``hive.orc.max-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``8 MB``
+ * **Description:** Serves as default value for ``orc_max_buffer_size`` session properties defining max size of ORC read operators. Higher value will allow bigger chunks to be processed but will decrease concurrency level.
+
+
+``hive.orc.max-merge-distance``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``1 MB``
+ * **Description:** Serves as default value for ``orc_max_merge_distance`` session property. Defines maximum size of gap between two reads to merge into a single read. The reads may be merged if distance between requested data ranges in data source is smaller or equal to this value.
+
+
+``hive.orc.stream-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``8 MB``
+ * **Description:** Serves as default value for ``orc_stream_buffer_size`` session property defining max size of ORC streaming operators. Higher value allows to load and process bigger chunks of data possibly increasing latency and decreasing parallelism but also increasing processing efficiency.
+
+
+``hive.orc.use-column-names``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Access ORC columns using names from the file. By default, columns in ORC files are accessed by their ordinal position in the Hive table definition. Setting this property allows to use columns names recorded in the ORC file instead.
```

Comment:
`By default, Hive access columns in ORC files using the order recoded in the Hive metastore`

Also, I'm not sure this works with files written by Hive since IIRC Hive does not store the columns names properly.